### ISSUE TITLE
Improve unit tests coverage for api code

### ISSUE THREAD

**Issue by ZibanPirate** [dzcode-io/dzcode.io#287]
**Description**

if you:
- go to `./api` folder
- then run `yarn test:cov:watch` inside it
- then open `./api/coverage/lcov-report/index.html` file in your browser

you will see that some directories are marked as red, which means test coverage is below 80%. you can also see the same results in [codecov.io](https://codecov.io/gh/dzcode-io/dzcode.io/tree/main/api/src)

In this task we will make sure that all dirs are marked green ✅.

**Check List**

1. [ ] add unit test for `./api/src/logger/service.ts`, similar to [config/service.spec.ts](https://github.com/dzcode-io/dzcode.io/blob/main/api/src/config/service.spec.ts)
2. [ ] add unit test for `./api/src/app/middlewares/error.ts`.
3. [ ] add unit test for `./api/src/app/middlewares/logger.ts`.
4. [ ] add unit test for `./api/src/app/middlewares/security.ts`.
5. [ ] add unit test for `./api/src/app/middlewares/docs.ts`.

after #286 is done, more tasks will be added here...

**Additional Comments**

Please create new PR for each task.

Also, it would be great for transparency if you comment below which task you want to take up.

**Comment by cherifGsoul** [comment]
The logger is an wrapper about winston I guess it should be covered with contract test.

**Comment by ZibanPirate** [comment]
well @cherifGsoul , since in this case the provider is a npm package (winstonre) which follows the semantic versioning, its API will always stay the same, and in case where we upgrade to a major version, then Typescript will report errors if we have API incompatibility.

Now that you brought this up, i'm thinking the [GithubService](https://github.com/dzcode-io/dzcode.io/blob/main/api/src/github/service.ts) needs a contract test don't you think? as it's does REST calls to Github APIs, if so, then we can create new **Task** ticket about contract testing where we define where we place them, be it `GithubService` or any other place

**Comment by cherifGsoul** [comment]
Semantic versioning doesn't guarantee that the library API will remain the same forever, major version allows breaking changes.

I would split GithHub service in two classes, the service itself just uses collaborators like the module that ensures communication with GitHub web services, the latter is candidate for contract tests.

That's my two cents, best of luck.

**Comment by cherifGsoul** [comment]
> and in case where we upgrade to a major version, then Typescript will report errors if we have API incompatibility.

The point of the wrapper is to be ready for the new API with minimum effort.
